Pages/Language_Page.cs

```diff
@@ -0,0 +1,230 @@
+        {
+            WaitHelpers.WaitToBeVisible(driver, "XPath", e_message, 3);
+            IWebElement errorMessage = driver.FindElement(By.XPath("//div[@class='ns-box-inner' and contains(text(), 'Please enter language and level')]"));
+            //IWebElement errorMessage = driver.FindElement(By.XPath("//div[contains(text(), 'Please enter language and level')]"));
+            return errorMessage.Text;
+            /*string ex_Message = "Please enter language and level";
+            Assert.That(errorMessage.Text == e_message, "Record has not been created. Test failed!");*/
+        }
+        /*public void Empty_ErrorMessage()
+        {
+            // Error message displayed
+            WebDriverWait wait = new WebDriverWait(driver, TimeSpan.FromSeconds(1));
+            wait.Until(SeleniumExtras.WaitHelpers.ExpectedConditions.ElementExists(By.XPath("//div[@class='ns-box-inner' and contains(text(), 'Please enter language and level')]")));
+            IWebElement errorMessage = driver.FindElement(By.XPath("//div[@class='ns-box-inner' and contains(text(), 'Please enter language and level')]"));
+            
+        }*/
+        /*public void GetErrorMessage_Existingdata()
+        {
+            // Error message displayed
+            WebDriverWait wait = new WebDriverWait(driver, TimeSpan.FromSeconds(10));
+            wait.Until(SeleniumExtras.WaitHelpers.ExpectedConditions.ElementExists(By.XPath("//div[@class='ns-box-inner' and contains(text(), 'This language is already exist in your language list.')]")));
+            IWebElement errorMessage = driver.FindElement(By.XPath("//div[@class='ns-box-inner' and contains(text(), 'This language is already exist in your language list.')]"));
+            string ex_Message = "This language is already exist in your language list.";
+            Assert.That(errorMessage.Text == ex_Message, "Record has not been created. Test failed!");
+        }*/
+
+        public string DuplicateData_ErrorMessage()
+        {
+
+            IWebElement errorMessage = driver.FindElement(By.XPath("//div[@class='ns-box-inner' and contains(text(), 'This language is already exist in your language list.')]"));
+            return errorMessage.Text;
+
+
+        }
+        
+        public void Update_Language(String language, String level)
+        {
+            WebDriverWait wait = new WebDriverWait(driver, TimeSpan.FromSeconds(10));
+            wait.Until(SeleniumExtras.WaitHelpers.ExpectedConditions.ElementIsVisible(By.XPath("//*[@id=\"account-profile-section\"]/div/section[2]/div/div/div/div[3]/form/div[2]/div/div[2]/div/table/tbody[last()]/tr/td[3]/span[1]/i")));
+            IWebElement edit_Button = driver.FindElement(By.XPath("//*[@id=\"account-profile-section\"]/div/section[2]/div/div/div/div[3]/form/div[2]/div/div[2]/div/table/tbody[last()]/tr/td[3]/span[1]/i"));
+            edit_Button.Click();
+            IWebElement edit_language = driver.FindElement(By.Name("name"));
```

Comment:
you need to think about state, you are think about clean up but not fully correctly

Reply:
see example before adding language you need to make sure there are no existing languages as this could result in false positive test pass
you will need to think about this for all the scenarios, this should be done before you start any tests 

the state is say that you have a test that is adding a language english, the second time you run this test you should delete any existing language so that the state is reset, this will guarantee that your test does not produce false negatives and passes always. You can not assume that test execution order is not always the same, imagine if you run the tests  in parallel mode the all tests will run at the same time so you need to also make sure you do not wipe out other tests data
hence why when you remove data that other tests are using.  The best thing to do is delete only data you added for the test to be correct
e.g test 1 added english and french as language test 2 added hindi then when you reset data you already know what language you are adding for each test
there for on reset of the two test you should delete english and french for test 1 and hindi for test 2

So to sum up you should delete all data before the start of any test and then after each test you should delete the data that a particular test is adding!
This is ensure that upon staring the test pack you have reset state and upon finishing each test deletes their data that they added!
Think about first clean up before any test runs (you delete all data around this feature e.g  before any language test run you delete all languages). Then after each test delete only data that the test has added! (test1 added english and test2 to added french then test1 should only delete english and test2 should delete french only) 

also say you are updating a language then you must added language then update it do not assume the language is there! This applies for delete or any scenario